packages/medusa/src/services/cart.ts

```diff
@@ -784,6 +851,17 @@ class CartService extends TransactionBaseService<CartService> {
           await this.updateShippingAddress_(cart, shippingAddress, addrRepo)
         }
 
+        if (
+          this.featureFlagRouter_.isFeatureEnabled(
+            SalesChannelFeatureFlag.key
+          ) &&
+          typeof data.sales_channel_id !== "undefined" &&
+          data.sales_channel_id != cart.sales_channel_id
+        ) {
+          await this.onSalesChannelChange(cart, data.sales_channel_id)
+          cart.sales_channel_id = data.sales_channel_id
+        }
+
```

Comment:
nit: to make it easier to clean up when we release the feature it might be better to do:

```suggestion
        if (this.featureFlagRouter_.isFeatureEnabled(SalesChannelFeatureFlag.key)) {
          if (
            typeof data.sales_channel_id !== "undefined" &&
            data.sales_channel_id != cart.sales_channel_id
          ) {
            await this.onSalesChannelChange(cart, data.sales_channel_id)
            cart.sales_channel_id = data.sales_channel_id
          }
        }

```